File: GetNewFile.py
import FnsEnginie
import settings
import os

def get_file_dif(base_file_list, new_filelist):
    a = set(line.strip() for line in open(base_file_list))
    b = set(line.strip() for line in open (new_filelist))
    c = b - a
    return c
    
def download_diff(fns_enginie: FnsEnginie.FnsDownload, set_to_download, target_dir):
    for f in set_to_download:
        fns_enginie.download_file(f, target_dir)

def download_new(fns_enginie: FnsEnginie.FnsDownload, base_file, new_file, target_dir):
    delta = get_file_dif(base_file, new_file)
    print(f'to download: {delta}')
    download_dif(fns_enginie, delta, target_dir)
    print('done to: ' + settings.FNS_dir_target)
    if os.path.isfile(base_file) and os.path.isfile(newfile):
        os.remove(base_file)
        os.remove(new_file, base_file)

if __name__ == '__main__' :
    print('GetNewFile started')
    fnsUL = FnsEnginie.FnsDownload(proxies = settings.proxies, headers = settings.headers,
            cert_str = settings.ulcert, key_srt = settings.ulkey, url = settings.url,
            start_dir = settings.ulstart_dir, ex_dir = settings.ex_dir)
    print('EGRUL enginie init done')
    fnsIP = FnsEnginie.FnsDownload(proxies = settings.proxies, headers = settings.headers,
            cert_str = settings.ipcert, key_srt = settings.ipkey, url = settings.url,
            start_dir = settings.ipstart_dir, ex_dir = settings.ex_dir)
    print('EGRIP enginie init done')
    fnsUL.file_list_to_file('EGRUL_last.txt')
    print('EGRUL_last.txt done')
    fnsIP.file_list_to_file('EGRIP_last.txt')
    print('EGRIP_last.txt done')
    download_new(fns_enfinie = fnsUL, base_file = 'EGRUL.txt', new_file = 'EGRUL_last.txt', target_dir = settings.FNS_dir_target)
    download_new(fns_enfinie = fnsIP, base_file = 'EGRIP.txt', new_file = 'EGRIP_last.txt', target_dir = settings.FNS_dir_target)
    print('ALL done')

File: GetNewFile.java
import java.util.HashSet;
import java.util.Set;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.IOException;
import FnsEnginie.FnsDownload;
import settings.Settings;

public class GetNewFile {
    public static Set<String> getFileDifference(String baseFileList, String newFileList) throws IOException {
        Set<String> baseFiles = new HashSet<>(Files.readAllLines(Paths.get(baseFileList)));
        Set<String> newFiles = new HashSet<>(Files.readAllLines(Paths.get(newFileList)));
        newFiles.removeAll(baseFiles);
        return newFiles;
    }

    public static void downloadDifferences(FnsDownload fnsEngine, Set<String> filesToDownload, String targetDir) {
        for (String file : filesToDownload) {
            fnsEngine.downloadFile(file, targetDir);
        }
    }

    public static void downloadNew(FnsDownload fnsEngine, String baseFile, String newFile, String targetDir) throws IOException {
        Set<String> delta = getFileDifference(baseFile, newFile);
        System.out.println("To download: " + delta);
        downloadDifferences(fnsEngine, delta, targetDir);
        System.out.println("Done to: " + Settings.FNS_dir_target);
        if (Files.isRegularFile(Paths.get(baseFile)) && Files.isRegularFile(Paths.get(newFile))) {
            Files.delete(Paths.get(baseFile));
            Files.delete(Paths.get(newFile));
        }
    }

    public static void main(String[] args) throws IOException {
        System.out.println("GetNewFile started");

        FnsDownload fnsUL = new FnsDownload(Settings.proxies, Settings.headers, Settings.ulcert,
                Settings.ulkey, Settings.url, Settings.ulstart_dir, Settings.ex_dir);
        System.out.println("EGRUL engine init done");

        FnsDownload fnsIP = new FnsDownload(Settings.proxies, Settings.headers, Settings.ipcert,
                Settings.ipkey, Settings.url, Settings.ipstart_dir, Settings.ex_dir);
        System.out.println("EGRIP engine init done");

        fnsUL.fileListToFile("EGRUL_last.txt");
        System.out.println("EGRUL_last.txt done");

        fnsIP.fileListToFile("EGRIP_last.txt");
        System.out.println("EGRIP_last.txt done");

        downloadNew(fnsUL, "EGRUL.txt", "EGRUL_last.txt", Settings.FNS_dir_target);
        downloadNew(fnsIP, "EGRIP.txt", "EGRIP_last.txt", Settings.FNS_dir_target);

        System.out.println("ALL done");
    }
}
